packages/nextjs/src/app-router/client/useAwaitableNavigate.ts

```diff
@@ -1,46 +1,61 @@
 'use client';
 
-import { useRouter } from 'next/navigation';
-import { useEffect, useRef, useTransition } from 'react';
+import { usePathname, useRouter } from 'next/navigation';
+import { useCallback, useEffect } from 'react';
 
-type NavigateFunction = ReturnType<typeof useRouter>['push'];
+declare global {
+  interface Window {
+    __internal_clerk_navPromisesBuffer: Array<() => void> | undefined;
+    __internal_clerk_navFun: (to: string) => Promise<void>;
+  }
+}
 
 /**
  * Creates an "awaitable" navigation function that will do its best effort to wait for Next.js to finish its route transition.
- *
  * This is accomplished by wrapping the call to `router.push` in `startTransition()`, which should rely on React to coordinate the pending state. We key off of
  * `isPending` to flush the stored promises and ensure the navigates "resolve".
  */
 export const useAwaitableNavigate = () => {
   // eslint-disable-next-line @typescript-eslint/unbound-method
   const { push } = useRouter();
-  const [isPending, startTransition] = useTransition();
-  const clerkNavRef = useRef<(...args: Parameters<NavigateFunction>) => Promise<void>>();
-  const clerkNavPromiseBuffer = useRef<(() => void)[]>([]);
+  const pathname = usePathname();
+
+  if (typeof window !== 'undefined') {
+    window.__internal_clerk_navFun = to => {
+      if (to === window.location.href.replace(window.location.origin, '')) {
```

Comment:
Why does this case need special handling?

Reply:
This is removed as I was testing snapshot versions